systemd: specify default-user-shell

andre4ik3 opened the pull request:
This option is used to specify the default shell used for homed users, among other things. By default it is set to `/bin/bash` which doesn't exist on NixOS.

Setting this option makes `homectl create` work without having to specify `--shell /run/current-system/sw/bin/bash`.

(I'm currently building the NixOS tests but it will take a while due to needing to rebuild the world in staging. I tested the change by setting `systemd.package` on unstable and it worked.)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ElvishJerricco reviewed (approved):
Incidentally, I was experimenting with exactly this a couple of weeks ago and discovered the need for the exact same fix. So I've already tested this weeks ago and found it good :P Forgot to open a PR though. Thanks!

Probably still need to run relevant NixOS tests though. I'll do that soon.

andre4ik3 committed:
systemd: specify default-user-shell

This option is used to specify the default shell used for homed users,
among other things. By default it is set to `/bin/bash` which doesn't
exist on NixOS.

Setting this option makes `homectl create` work without having to
specify `--shell /run/current-system/sw/bin/bash`.

andre4ik3 commented:
I did a force-push with the same commit contents, just a different timestamp to trigger the CI to run again.
